### ISSUE TITLE
First level design

### ISSUE THREAD

**Issue by BojanStankovic** [PlaceholderGames/2016-VnD_Game#26]
A basic level must be designed which will demonstrate all the game mechanics that is already implemented (or is about to be implemented).

It must include a win/lose condition

**Comment by BojanStankovic** [comment]
It would be best if everyone comes up with an idea for a level by 02.02.2017. Do what you can, it doesn't have to be perfect or implement everything we mentioned until now. When we compare individual ideas we can take bits and pieces from each one and create a single decent level (perhaps even two).

**Comment by rbakerusw** [comment]
Ok, I will try to come up with some ideas. The only issue right now is that we don't have many features for the ants to do so it is hard to think of a level when we don't know what features will be ready by then. Realistically, what features are we aiming for by then?

**Comment by ghost** [comment]
Ok I'll draw up a small concept level and bring it to class on Thursday to show you guys! :)

**Comment by ghost** [comment]
Here's some preliminary ideas, I know its 2 dimensional but I had no idea how to draw a 3D level, But the ideas are transferable, While doing this I think for later on the ability to stop ants in their track and make them carry on again would be useful?

The ants spawn at the beginning of the level and make their way to the finish point overcoming all obstacles. Hopefully the Key should explain all tiles.

![image](https://cloud.githubusercontent.com/assets/22819171/23029016/011bf6c2-f461-11e6-84cb-caad92f5e7dc.png)

**Comment by ghost** [comment]
For now we can jut let the ants fall down the holes but later on maybe the flying / gliding skill can be used to safely make the land? :)

**Comment by BojanStankovic** [comment]
As we discussed of Facebook I have created a branch for the level design for each of us. Branches are named I26_YOUR-NAME_first_level_design. You can name your level FirstLevel_YOUR-NAME so we don't have merge conflicts and we can potentially keep all the levels

**Comment by elliotnaylor** [comment]
We can't exactly start making the levels because we're missing direction change nodes.  This is the main mechanic and it would be pointless to make a level without it

**Comment by BojanStankovic** [comment]
And what are the nodes in the test level being used right now? They seem like they are changing the direction

**Comment by elliotnaylor** [comment]
But we need the ants to be able to place them

**Comment by BojanStankovic** [comment]
We need many things still, however not all of them can be implemented for the final deadline, let alone for the 1st build. What we do need for the first build is a demonstration of what have we currently been working on and that requires some kind of a level with at least minimal interaction (like mining and digging). If you don't feel like creating a level that is perfectly fine. You can focus on something else. Plenty of bugs are still crawling around 😃

**Comment by elliotnaylor** [comment]
Yeah I see your point.  We already have the method for spawning objects from an ant, should I implement it into a tool just so we can make some better levels?

**Comment by napiorek** [comment]
It's hard to come up with something interesting if we only got two features that the ants can use, hence here is what I come up with. Initially my idea was to create the spiral stairs where ants could climb up,but very quickly I realized that our camera could not follow the ants as it's attached to 3rd person 👎 
any way, here is a quick demo, ver2 showing what would happened if player accidently highlight additional ant and  trigger digging ( the following anst would fall down and would have to go around to come back to previous position in order to continue the run to the finish)

Version 1  https://youtu.be/vqPT4gxoWz0 
Version 2  https://youtu.be/BcPExzBFb0Y 

(All uploaded to my branch)

**Comment by BojanStankovic** [comment]
Yeah, we should probably remove the body for the ThirdPersonCharacter so it can move more freely around the map. I see what you mean by having a hard time coming up with a level. Additionally I'm having issues with the box that is removed on digging/mining. It happens that the ants just fall through it. I'll open an issue for it

**Comment by napiorek** [comment]
First build done and presented in class. All conclusions are being discuses in the issue #52  Important: Second build.

**Comment by BojanStankovic** [comment]
@napiorek The branch related to this issue was not merged and it was not supposed to be closed until merged.

**Comment by BojanStankovic** [comment]
I merged Ryan's level into the develop branch. I believe that a bug with ants passing through the blocks was solved on this branch, but I didn't want to introduce those changes to reduce the chance of braking the project. If I remember correctly it was a small change but I don't know on what exactly and Elliot told me he did some other changes that shouldn't be merged so I took the safest route. Develop tested after the merge. Closing
